### ISSUE TITLE
Diablo game not accepting some inputs

### ISSUE THREAD

**Issue by jlfreund** [libretro/swanstation#58]
I have a weird problem where Diablo running on my Retroid Pocket 3 (Android) using Swanstation was working perfectly, but would occasionally get key input problems.  I'm currently stuck on the Diablo "Controller Setup" screen and almost all keys work there for navigating the menu, changing key mappings, etc.  But the game does not recognize "start" or "select" on this screen to exit the menu back to the game, and even stranger, the triangle button required to go "Back" to the previous menu only works when used to change a control mapping to the triangle, but when focusing on any other control on the left column of that menu, the triangle isn't recognized, so I can't go back to the previous menu.   All the keys work in other PSX games, and normally work in Diablo as well, except in this one "Controller Setup" screen.

For my DeviceType, (Settings -> Input -> Port 1 Controls -> Device Type), I have it set to "Analog Controller (Dual Shock)", but have tried all other menu options there.

**Comment by jlfreund** [comment]
Note, several Reddit complaints about this issue as well:
- [Reddit 1](https://www.reddit.com/r/RetroPie/comments/dhkivg/comment/f3p6lf0/): Diablo on PSX not accepting input
- [Reddit 2](https://www.reddit.com/r/RetroArch/comments/k9i8x8/does_anyone_know_how_to_fix_the_controls_not/): Diablo controller issue
- [Reddit 3](https://www.reddit.com/r/RetroPie/comments/5qvs87/cant_get_the_start_button_to_work_on_playstation/): Start button in Diablo

**Comment by DarthMew** [comment]
Seems to be more a 'problem' with the game, as when any of the important inputs like attack or to cast spells get unmapped, it will refuse to let you exit the menu until you remap them to a valid button.

With default ingame settings, and default RetroPad settings for both the digital and analog (DualShock) controller profiles, the game works fine.

**Comment by jlfreund** [comment]
Yes, but I don't think it was a problem in the original PS1 game, just when emulating.  I'm still experimenting with other settings and emulators, but so far, can only reproduce reliably with Swanstation (not with Beetle)

**Comment by DarthMew** [comment]
To make absolutely sure, when I boot up the game, and from the title screen go into the options and select the Controller Setup section, when I haven't messed with any of the input settings, with both SwanStation and Beetle, I can exit out when pressing triangle.
As soon as either "Attack", "Action" or "Cast Spell" get unmapped for whatever reason, I can't exit until out until I have the all three mapped, again on both SwanStation and Beetle.
And no matter what, with neither SwanStation nor Beetle, start and select can't be mapped (which makes sense, as start is used to pause, and select allows you to access the actual pause menu when the game is paused) nor will allow you to exit out of the submenu.

Is any of what I described not supposed to happen, because what I've experienced sure does feel intentional to me.
